### PR TITLE
chip away at Topic/jbuilder items

### PR DESCRIPTION
**Pull request by gtrak**
Fixes the duplicate constructors by disregarding the 'shape' attribute during errors sort.

Removes .envrc (I noticed this had been .gitignored prior)

Removes unused lib makefile.

Fixes oUnit dep in lib_test.

Adds a compile-libraries make target (maybe there's a better way to do this?).

Fixed the RDS compiler error by properly handling empty structs.
Fixed the QuotaExceeded by using Util.variant_name.

```
File "lib/createDBSecurityGroup.ml", line 53, characters 4-40:
Error: Unbound module Errors.QuotaExceeded
```

### CODE REVIEW

File: libraries/autoscaling/lib/types.ml [hunk 1]
```diff
@@ -2622,6 +2622,15 @@ module ExitStandbyQuery =
              (Util.of_option_exn (Json.lookup j "auto_scaling_group_name")))
       } 
   end
+module RecordLifecycleActionHeartbeatAnswer =
+  struct
+    type unit
```
Comment (gtrak) [inline libraries/autoscaling/lib/types.ml]: I'll look into making this prettier, seems like it compiles?

**Comment by gtrak** [comment]
I plan to remove the gen'd code from this PR

**Comment by gtrak** [comment]
gen'd RDS code looks like this now:
```
open Types
open Aws
type input = unit
type output = AccountAttributesMessage.t
type error = Errors.t
let service = "rds" 
let to_http req =
  let uri =
    Uri.add_query_params (Uri.of_string "https://rds.amazonaws.com")
      (List.append
         [("Version", ["2014-10-31"]);
         ("Action", ["DescribeAccountAttributes"])]
         (Util.drop_empty (Uri.query_of_encoded "")))
     in
  (`POST, uri, []) 
```

**Comment by gtrak** [comment]
Is there anything we can do about the trailing whitespace in all the gen'd code? Is that a migrate-parsetree thing?

**Comment by gtrak** [comment]
I made it a little cleaner..

```
open Types
open Aws
type input = unit
type output = AccountAttributesMessage.t
type error = Errors.t
let service = "rds" 
let to_http req =
  let uri =
    Uri.add_query_params (Uri.of_string "https://rds.amazonaws.com")
      [("Version", ["2014-10-31"]);
      ("Action", ["DescribeAccountAttributes"])]
     in
  (`POST, uri, []) 
```

**Comment by tmcgilchrist** [comment]
Nice work, thank you.

**Pull request merged**